src/java.base/share/classes/java/lang/classfile/package-info.java

```diff
@@ -242,25 +241,49 @@
  * <p>
  * If a constant pool entry has a nominal representation then it provides a
  * method returning the corresponding nominal descriptor type e.g.
- * method {@link jdk.internal.classfile.constantpool.ClassEntry#asSymbol} returns
+ * method {@link java.lang.classfile.constantpool.ClassEntry#asSymbol} returns
  * {@code ClassDesc}.
  * <p>
  * Where appropriate builders provide two methods for building an element with
  * symbolic information, one accepting nominal descriptors, and the other
  * accepting constant pool entries.
  *
+ * <h3>Consistency checks, syntax checks and verification</h3>
+ * No consistency checks are performed while building or transforming classfiles
+ * (except for null arguments checks). All builders and classfile elements factory
+ * methods accepts provided information without implicit validation.
+ * However fatal inconsistencies (like for example invalid code sequence or
+ * unresolved labels) affects internal tools and may cause exceptions later in
+ * the classfile building process.
+ * <p>
+ * Basic syntax control can be achieved using symbolic descriptors. For example
+ * using {@link java.lang.constant.ClassDesc} assures the right serial form is
+ * used in the particular place inside classfile and also assures the class name
+ * has been already syntactically validated.
+ * <p>
+ * On the other hand it is possible to use builders methods and factories accepting
+ * constant pool entries to avoid any form of syntax control. Constant pool entries
+ * can be constructed from raw values, where no syntax control is in charge.
+ * {@snippet lang=java :
+ * var invalidInternalClassName = constantPoolBuilder.classEntry(
+ *                                    constantPoolBuilder.utf8Entry("mypackage.MyClass"));
+ * }
+ * <p>
+ * More complex verification of a classfile can be achieved by explicit invocation
+ * of {@link java.lang.classfile.ClassModel#verify}.
```

Comment:
Aren't part of verification also ran as part of generating stackmaps (unless stackmap inference is disabled) ? Should it be mentioned here?

Reply:
Only fragment of verification conditions are required for stackmaps generator and it is indirectly mentioned in the sentence: "However fatal inconsistencies (like for example invalid code sequence or unresolved labels) affects internal tools and may cause exceptions later in the classfile building process."